tasks/release.py

```diff
@@ -0,0 +1,59 @@
+"""
+Release helper tasks
+"""
+from __future__ import print_function
+
+from invoke import task, Failure
+
+
+@task
+def update_changelog(ctx, new_version):
+    """
+    Quick task to generate the new CHANGELOG using reno when releasing a minor
+    version.
+    """
+    new_version_int = map(int, new_version.split("."))
+
+    if len(new_version_int) != 3:
+        print("Error: invalid version: {}".format(new_version_int))
+        raise Exit(1)
+
+    # let's check that the tag for the new version is present (needed by reno)
+    try:
+        ctx.run("git tag --list | grep {}".format(new_version))
+    except Failure as e:
+        print("Missing '{}' git tag: mandatory to use 'reno'".format(new_version))
+        raise
+
+    # make sure we are up to date
+    ctx.run("git fetch")
+
+    # removing releasenotes from bugfix on the old minor.
+    previous_minor = "%s.%s" % (new_version_int[0], new_version_int[1] - 1)
+    ctx.run("git rm `git log {}.0...remotes/origin/{}.x --name-only \
+            | grep releasenotes/notes/`".format(previous_minor, previous_minor))
+
+    # generate the new changelog
+    ctx.run("echo 'reno report \
+            --ignore-cache \
+            --earliest-version {}.0 \
+            --version {} \
+            --no-show-source > /tmp/new_changelog.rst'".format(previous_minor, new_version))
+    ctx.run("reno report \
+            --ignore-cache \
+            --earliest-version {}.0 \
+            --version {} \
+            --no-show-source > /tmp/new_changelog.rst".format(previous_minor, new_version))
+
+    # reseting git
+    ctx.run("git reset --hard HEAD")
```

Comment:
since we do a `git reset --hard` here, I think it's nicer to check that `git diff HEAD` doesn't return anything at the beginning of the task, and fail the command early if there's something in the diff so that the user doesn't accidentally throw away uncommitted code. `git diff --exit-code HEAD` seems to be the command to use :)

Reply:
Good catch ! I was thinking about running this task in our CI where it's safe. But in case we want to use it locally it's a must have !